### ISSUE TITLE
Is there a way to check if a subscription was unsubscribed to using a mock subscription?

### ISSUE THREAD

**Issue by danieltian** [Mike-Gibson/mock-apollo-client#53]
I'm using [Vue Apollo's smart subscription](https://apollo.vuejs.org/api/smart-subscription.html) that I'm manually calling `start()` and `stop()` on. In the tests, I created a mock subscription using `createMockSubscription()`:

```js
const mockSubscription = createMockSubscription()
```

I can test if the subscription starts by checking if `mockSubscription.closed` changes from `true` to `false` after the subscription is started:

```js
expect(mockSubscription.closed).toBe(true)
doSomethingThatMakesSubscriptionStart()
expect(mockSubscription.closed).toBe(false)
```

but when I try to test if the subscription stops, `mockSubscription.closed` does not change from `false` to `true`:

```js
doSomethingThatMakesSubscriptionStart()
expect(mockSubscription.closed).toBe(false)
doSomethingThatMakesSubscriptionStop()
expect(mockSubscription.closed).toBe(true) // fails
```

It looks like the only way `mockSubscription.closed` can be changed from `false` to `true` is to call `mockSubscription.complete()`, but that manually stops the mock subscription, whereas I need to verify that my code is stopping the actual subscription. Is there any way to do this using `mock-apollo-client`?

**Comment by Mike-Gibson** [comment]
Thanks for the question.

Unfortunately, I don't think this is currently possible with mock apollo client as it stands.

I hadn't seen the need for it previously, as I've only seen subscriptions being used via React hooks, which automatically handle subscribing and unsubscribing.

I took a quick look and it looks like the smart subscription will call `stop` automatically when the component is unmounted. I'm interested to know - in your case is manually calling stop required for a specific use case?

I've had a quick look and I think this should be possible to implement, but I'm still thinking about it, as the mock subscription functionality has been implemented in a very basic way, and I don't think it would handle more advanced scenarios (like subscriptions for the same query multiple times) and wonder whether it could do with improving before adding this functionality.
